### ISSUE TITLE
compile error on vonnegut

### ISSUE THREAD

**Issue by evanmcc** [ferd/rebar3_proper#18]
Tried debugging some, but quickly got out of my depth.  Here's the stack trace and rebar3 report:

```
===> Compiling vonnegut for PropEr
===> Compiling files in /Users/evan/src/vonnegut2/test to /Users/evan/src/vonnegut2/_build/test/lib/vonnegut/test
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace or consult rebar3.crashdump
===> Uncaught error: function_clause
===> Stack trace to the error location:
[{filename,join,[[]],[{file,"filename.erl"},{line,399}]},
 {rebar_dir,'-src_dirs/3-lc$^0/1-0-',1,
            [{file,"/home/tristan/Devel/rebar3/_build/default/lib/rebar/src/rebar_dir.erl"},
             {line,264}]},
 {rebar_dir,src_dirs,3,
            [{file,"/home/tristan/Devel/rebar3/_build/default/lib/rebar/src/rebar_dir.erl"},
             {line,261}]},
 {rebar_erlc_compiler,compile,4,
                      [{file,"/home/tristan/Devel/rebar3/_build/default/lib/rebar/src/rebar_erlc_compiler.erl"},
                       {line,151}]},
 {rebar3_proper_prv,'-compile_dirs/3-lc$^0/1-0-',3,
                    [{file,"/Users/evan/src/vonnegut2/_build/default/plugins/rebar3_proper/src/rebar3_proper_prv.erl"},
                     {line,245}]},
 {rebar3_proper_prv,compile_dirs,3,
                    [{file,"/Users/evan/src/vonnegut2/_build/default/plugins/rebar3_proper/src/rebar3_proper_prv.erl"},
                     {line,241}]},
 {rebar3_proper_prv,find_properties,4,
                    [{file,"/Users/evan/src/vonnegut2/_build/default/plugins/rebar3_proper/src/rebar3_proper_prv.erl"},
                     {line,210}]},
 {rebar3_proper_prv,find_properties,2,
                    [{file,"/Users/evan/src/vonnegut2/_build/default/plugins/rebar3_proper/src/rebar3_proper_prv.erl"},
                     {line,186}]}]
```

```
$ rebar3 report proper
===> Compiling rebar3_proper
Rebar3 report
 version 3.4.4
 generated at 2017-11-10T18:06:20+00:00
=================
Please submit this along with your issue at https://github.com/erlang/rebar3/issues (and feel free to edit out private information, if any)
-----------------
Task: proper
Entered as:
  proper
-----------------
Operating System: x86_64-apple-darwin15.6.0
ERTS: Erlang/OTP 20 [erts-9.1.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:0] [hipe] [kernel-poll:false] [dtrace]
Root Directory: /usr/local/Cellar/erlang/20.1.2/lib/erlang
Library directory: /usr/local/Cellar/erlang/20.1.2/lib/erlang/lib
-----------------
Loaded Applications:
bbmustache: 1.3.0
certifi: 2.0.0
cf: 0.2.2
common_test: 1.15.2
compiler: 7.1.3
crypto: 4.1
cth_readable: 1.3.0
dialyzer: 3.2.2
edoc: 0.9.1
erlware_commons: 1.0.0
eunit: 2.3.4
eunit_formatters: 0.3.1
getopt: 0.8.2
inets: 6.4.2
kernel: 5.4
providers: 1.6.0
public_key: 1.5
relx: 3.23.1
sasl: 3.1
snmp: 5.2.7
ssl_verify_fun: 1.1.2
stdlib: 3.4.2
syntax_tools: 2.1.3
tools: 2.11

-----------------
Escript path: undefined
Providers:
  app_discovery as clean compile compile cover ct deps dialyzer do edoc escriptize eunit get-deps help install install_deps list lock new path pkgs proper release relup report run send shell state tar tree unlock update upgrade upgrade upgrade version xref ```

**Comment by ferd** [comment]
That is fixed on the master version of rebar3. I introduced a bug in 3.4.4. We're due for a new rebar3 release.

**Comment by evanmcc** [comment]
this is fixed in 3.4.6
